single.c

```diff
@@ -3,14 +3,16 @@
 #include "brute.h"
 #include <string.h>
 
+#define crypt_r __crypt_r
```

Comment:
Better to use weak_alias(__crypt_r, crypt_r);

Reply:
> Better to use weak_alias(__crypt_r, crypt_r);

There is no `weak_alias` on MacOS, I tried to change it to its define on Linux (`extern __typeof(__crypt_r) crypt_r __attribute__((weak, alias("__crypt_r")));` and got `error: aliases are not supported on darwin`. Are there any alternatives?